webapi/src/main/java/com/dumbdogdiner/stickyapi/util/http/UrlValidator.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (c) 2020-2021 DumbDogDiner <dumbdogdiner.com>. All rights reserved.
+ * Licensed under the MIT license, see LICENSE for more information...
+ */
+package com.dumbdogdiner.stickyapi.util.http;
+
+import lombok.Getter;
+import okhttp3.HttpUrl;
+
+import java.net.URL;
+import java.util.StringJoiner;
+import java.util.regex.Pattern;
+
+/**
+ * A URL Validator, based on <a href="https://owasp.org/www-community/OWASP_Validation_Regex_Repository">OWASP's URL
+ * regex</a>, with minor tweaks to remove unsupported protocols and allow a domain to be specified
+ */
+public class UrlValidator {
+    /**
+     * The regex used by default to test if a domain is valid
+     */
+    public static final Pattern DEFAULT_DOMAIN_REGEX = Pattern.compile("(%[0-9A-Fa-f]{2}|[-()_.!~*';/?:@&=+$,A-Za-z0-9])+");
+
+    @Getter
+    private static final UrlValidator instance = new UrlValidator();
+
+    /**
+     * The regex used by default to test if URLs are valid
+     */
+    @SuppressWarnings("unused")
+    public static final Pattern DEFAULT_URL_REGEX = instance.urlRegex;
+
+    @Getter
+    private final Pattern urlRegex;
+
+    private UrlValidator() {
+        this(DEFAULT_DOMAIN_REGEX);
+    }
+
+    private UrlValidator(Pattern domain){
```

Comment:
Yes, this is a private method, but for future devs can you comment on what it does?